### PR TITLE
Mes 4900 state management setup

### PR DESCRIPTION
**Pull request by mgoldson**
## Description
Adds home categories(F,G,H,K) to the Reducer Factory, including reducer composition of:

 - Tests
 - Test Data
 - Journal Data
 - Vehicle Details

state for Vehicle Checks and Highway Safety Code Reducers currently return null and reducers will be completed in specific stories associated with that functionality.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/modules/tests/test-data/cat-home/controlled-stop/__tests__/controlled-stop.reducer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { CatFUniqueTypes } from '@dvsa/mes-test-schema/categories/F';
```
Comment (TomBillingtonUK) [inline src/modules/tests/test-data/cat-home/controlled-stop/__tests__/controlled-stop.reducer.spec.ts:1]: Can we not have a common controlled stop reducer for B and Home , objects are the same so it should be easy to achive.

Comment (TomBillingtonUK) [reply src/modules/tests/test-data/cat-home/controlled-stop/__tests__/controlled-stop.reducer.spec.ts:1]: Happy for this to be chucked into the tech debt list - to get this in though

File: src/modules/tests/vehicle-details/cat-home/vehicle-details.cat-home.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as vehicleDetailsActions from '../common/vehicle-details.actions';
+import { VehicleDetails } from '@dvsa/mes-test-schema/categories/common';
+import { createFeatureSelector } from '@ngrx/store';
```
Comment (TomBillingtonUK) [inline src/modules/tests/vehicle-details/cat-home/vehicle-details.cat-home.reducer.ts]: this should be using a common one as it's a common type

**Review by TomBillingtonUK — APPROVED** [review]
Overall looks good, I think there is some duplication which can be removed , it can be in this PR or a second follow up depending on how urgently we need this.

File: src/modules/tests/test-data/cat-home/controlled-stop/__tests__/controlled-stop.reducer.spec.ts (hunk 1 above)
Comment (mgoldson) [reply src/modules/tests/test-data/cat-home/controlled-stop/__tests__/controlled-stop.reducer.spec.ts:1]: Yeah, good shout. I will do it as a separate PR though as we would making a change to category in prod. Better traceability that way.

**Review by faggc — APPROVED** [review]

**Pull request merged**